src/rock-gazebo.in

```diff
@@ -1,4 +1,14 @@
-#! /bin/sh
+#! /usr/bin/ruby
+require 'rock/bundles'
+require 'optparse'
 
-gazebo --verbose -s @CMAKE_INSTALL_PREFIX@/lib/librock_gazebo.so "$@"
+options = OptionParser.new
+args = options.parse(ARGV)
+world_file = args.shift
 
+Bundles.load
+
+ENV['GAZEBO_MODEL_PATH']=(ENV['GAZEBO_MODEL_PATH']||"")+
+Bundles.find_dirs('data','gazebo','models', :all => true, :order => :specific_first).join(":")
+
+Process.exec("gazebo --verbose -s  @CMAKE_INSTALL_PREFIX@/lib/librock_gazebo.so #{world_file}")
```

Comment:
This is generally fine, but there's a better way to use Process.exec:

Call Process.exec with a list of strings instead of a big string. In the latter case, the string gets executed with a subshell *and* therefore the arguments are re-parsed. This is fragile, as you have to be super-careful not to do anything the shell could interpret (as e.g. having a $ or a space in the strings you are interpolating). The multi-string case directly passes the arguments to the process, which avoids these problems.

It is possible to override the environment in Process.exec (you'll have to look at the documentation ...). This is very cosmetic in this particular case, but has a much better feeling to it ...